src/protocol/codecs/topic.rs

```diff
@@ -0,0 +1,112 @@
+use std::io::{self, ErrorKind};
+
+use bytes::BytesMut;
+use tokio_util::codec::Decoder;
+
+use crate::protocol::invalid_data;
+
+/// Topic is a key-value string pair.
+pub struct Topic {
+    /// Key.
+    pub key: String,
+
+    /// Value.
+    pub value: String,
+}
+
+#[derive(Default, Clone)]
+pub struct TopicCodec {}
+
+impl Decoder for TopicCodec {
+    type Item = Vec<Topic>;
+    type Error = io::Error;
+
+    fn decode(&mut self, src: &mut BytesMut) -> Result<Option<Self::Item>, Self::Error> {
+        if src.is_empty() {
+            return Ok(Some(vec![]));
+        }
+
+        let (&num_topics, mut src) = src
+            .split_first()
+            .ok_or_else(|| invalid_data!("couldn't decode the number of topics"))?;
+        let mut topics = Vec::with_capacity(num_topics as usize);
+
+        for _ in 0..num_topics {
+            let (&key_len, buf) = src
+                .split_first()
+                .ok_or_else(|| invalid_data!("couldn't decode the key length"))?;
+            if key_len as usize > buf.len() {
```

Comment:
```suggestion
                .ok_or_else(|| invalid_data!("couldn't decode the key length"))?;

            if key_len as usize > buf.len() {
```
just for consistency with L48